pkg/destroy/aws/aws.go

```diff
@@ -77,1510 +63,1205 @@ func (o *ClusterUninstaller) validate() error {
+		GroupId: aws.String(id),
+	})
+	if err != nil {
+		if err.(awserr.Error).Code() == "InvalidGroup.NotFound" {
+			return nil
 		}
+		return err
 	}
 
+	logger.Info("Deleted")
 	return nil
 }
 
-// deleteInstanceProfile will attempt to delete the provided instance profile
-func deleteInstanceProfile(instanceProfileID *string, iamClient *iam.IAM, logger logrus.FieldLogger) error {
-	ipList, err := iamClient.ListInstanceProfiles(&iam.ListInstanceProfilesInput{})
+func deleteEC2Subnet(client *ec2.EC2, id string, logger logrus.FieldLogger) error {
+	_, err := client.DeleteSubnet(&ec2.DeleteSubnetInput{
+		SubnetId: aws.String(id),
+	})
 	if err != nil {
-		logger.Debugf("error listing instance profiles: %v", err)
-		return err
-	}
-
-	var matchedIP *iam.InstanceProfile
-	for _, ip := range ipList.InstanceProfiles {
-		if *ip.InstanceProfileId == *instanceProfileID {
-			matchedIP = ip
+		if err.(awserr.Error).Code() == "InvalidSubnetID.NotFound" {
+			return nil
 		}
+		return err
 	}
 
-	if matchedIP == nil {
-		// nothing found, so already deleted?
-		return nil
-	}
-
-	// first delete any roles out of the instance profile
-	err = deleteRolesFromInstanceProfile(matchedIP, iamClient, logger)
-	if err != nil {
-		return errors.Errorf("error deleting roles from instance profile: %v", err)
-	}
+	logger.Info("Deleted")
+	return nil
+}
 
-	logger.Debugf("deleting instance profile: %v", *matchedIP.InstanceProfileName)
-	_, err = iamClient.DeleteInstanceProfile(&iam.DeleteInstanceProfileInput{
-		InstanceProfileName: matchedIP.InstanceProfileName,
+func deleteEC2Volume(client *ec2.EC2, id string, logger logrus.FieldLogger) error {
+	_, err := client.DeleteVolume(&ec2.DeleteVolumeInput{
+		VolumeId: aws.String(id),
 	})
 	if err != nil {
-		logger.Debugf("error deleting instance profile: %v", err)
+		if err.(awserr.Error).Code() == "InvalidVolume.NotFound" {
+			return nil
+		}
 		return err
-	} else if err == nil {
-		logger.WithField("name", *matchedIP.InstanceProfileName).Info("Deleted instance profile")
 	}
 
+	logger.Info("Deleted")
 	return nil
 }
 
-// tryDeleteRoleProfileByName attempts to delete roles and profiles with given name ($CLUSTER_NAME-bootstrap|master|worker-role|profile)
-func tryDeleteRoleProfileByName(roleName string, profileName string, session *session.Session, logger logrus.FieldLogger) error {
-	logger.Debugf("deleting role: %s", roleName)
-	describeRoleInput := iam.GetRoleInput{}
-	describeRoleInput.RoleName = &roleName
-	iamClient := iam.New(session)
-	if _, err := iamClient.GetRole(&describeRoleInput); err != nil && err.(awserr.Error).Code() != iam.ErrCodeNoSuchEntityException {
-		return err
+func deleteEC2VPC(ec2Client *ec2.EC2, elbClient *elb.ELB, elbv2Client *elbv2.ELBV2, id string, logger logrus.FieldLogger) error {
+	// first delete any Load Balancers under this VPC (not all of them are tagged)
+	v1lbError := deleteElasticLoadBalancerClassicByVPC(elbClient, id, logger)
```

Comment:
This is different from https://github.com/openshift/installer/blob/d1272424396f1d42faf239dc1b976db905d0b338/pkg/destroy/aws/aws.go#L463-L464

Reply:
> This is different from...

Is it?  I'm trying to repeat the earlier "try to delete both classic and v2 load balancers, regardless of failure, but error out if either failed".  Do you read this or the previous code as doing something else?

Reply:
> err nil continues ** this case is different.

Ah, good catch.  I've pushed a6e65d3 -> 247e1ec, so now it errors out too.